api/document/json_importer/importer.py

```diff
@@ -0,0 +1,131 @@
+from collections import defaultdict
+from typing import (Callable, DefaultDict, Dict, List, NamedTuple, Optional,
+                    Type)
+
+from django.db import transaction
+
+from document.models import Annotation, DocNode, ExternalLink, FootnoteCitation
+from document.tree import JSONAwareCursor, JsonDict
+from reqs.models import Policy
+
+
+class PreparedNode(NamedTuple):
```

Comment:
In the original prototype code this was called `CleanedNode`, but I think `PreparedNode` is a better name--"cleaned" suggests some sort of validation, which isn't being done here (validation is explicitly done by the DRF serializers).

Reply:
Without digging in _too_ deep, it feels like this tuple is duplicating `DocCursor`'s interface. I wonder if there's a more one-to-one way to do this conversion.